src/pages/workspace/companyCards/WorkspaceCompanyCardsList.tsx

```diff
@@ -0,0 +1,252 @@
+                            cardName={cardName}
+                            selectedFeed={selectedFeed}
+                            plaidIconUrl={plaidIconUrl}
+                            customCardName={customCardName}
+                            isHovered={hovered}
+                            isAssigned={!!assignedCard}
+                            onAssignCard={() => onAssignCard(cardIdentifier)}
+                            isAssigningCardDisabled={isAssigningCardDisabled}
+                            shouldUseNarrowTableRowLayout={shouldUseNarrowTableRowLayout}
+                        />
+                    )}
+                </PressableWithFeedback>
+            </OfflineWithFeedback>
+        );
+    };
+
+    const keyExtractor = (item: string, index: number) => `${item}_${index}`;
+
+    const ListHeaderComponent = shouldUseNarrowTableRowLayout ? (
+        <View style={styles.h7} />
+    ) : (
+        <>
+            {(cards?.length ?? 0) > CONST.SEARCH_ITEM_LIMIT && (
+                <SearchBar
+                    label={translate('workspace.companyCards.findCard')}
+                    inputValue={inputValue}
+                    onChangeText={setInputValue}
+                    shouldShowEmptyState={isSearchEmpty}
+                    style={[styles.mt5]}
+                />
+            )}
+            {!isSearchEmpty && (
+                <View style={[styles.flexRow, styles.appBG, styles.justifyContentBetween, styles.mh5, styles.gap5, styles.p4]}>
+                    <View style={[styles.flex1]}>
+                        <Text
+                            numberOfLines={1}
+                            style={[styles.textMicroSupporting, styles.lh16]}
+                        >
+                            {translate('common.member')}
+                        </Text>
+                    </View>
+                    <View style={[styles.flex1]}>
+                        <Text
+                            numberOfLines={1}
+                            style={[styles.textMicroSupporting, styles.lh16]}
+                        >
+                            {translate('workspace.companyCards.card')}
+                        </Text>
+                    </View>
+                    <View style={[styles.flex1]}>
+                        <Text
+                            numberOfLines={1}
+                            style={[styles.textMicroSupporting, styles.textAlignRight, styles.lh16, styles.pr7]}
+                        >
+                            {translate('workspace.companyCards.cardName')}
+                        </Text>
+                    </View>
+                </View>
+            )}
+        </>
+    );
+
+    // Show empty state when there are no cards
+    if (!cards?.length && !isLoadingCardsTableData) {
+        return (
+            <WorkspaceCompanyCardsFeedAddedEmptyPage
+                shouldShowGBDisclaimer={shouldShowGBDisclaimer}
+                handleAssignCard={onAssignCard}
```

Comment:
All other places in this PR the code is `andleAssignCard={() => onAssignCard()}`. Is this OK like this? I assume it's passing a cardID to the callback and the other references are specifically NOT passing a cardID?